### ISSUE TITLE
"Independent" IP inversion in chargeability units.

### ISSUE THREAD

**Issue by Santi-Rebole-Canals** [gimli-org/gimli#773]
Hello,

I am currently trying to invert an ERT res2dinv file with IP (in time-domain). I would like to know if there is an option to invert only IP data (can be done independently from resistivity data?).
As I can see, when inverting IP data using the "invert" function of ERTIPManager, two inversions are performed consecutively. However, when the results are represented, they are displayed in resistivity units rather than chargeability units (time-domain IP, in miliseconds). 
How can I invert the IP data alone? Additionally, how can I obtain an inverted tomography of IP in chargeability units?

Here is an image of the final results:

![Captura](https://github.com/user-attachments/assets/a967824b-31a6-4ae9-b079-1a02f2531abf)

Thank you for your assistance.

**Comment by halbmy** [comment]
IP data can never be inverted independent on DC data, that's the nature of the physics.

What you show seems to be the resistivity. Calling `mgr.showIPModel` should diet the IP model in chargeability units. See also the example https://www.pygimli.org/_examples_auto/3_induced_polarization/plot_08_TDIPinversion.html

**Comment by Santi-Rebole-Canals** [comment]
Thank you very much, I could obtain the results separately.
However, I have another question that I would like to clarify:

When using ERTManager for inverting and obtaining the resistivity image, I get one result. But when I use ERTIPManager for inverting and obtaining both resistivity and IP images, the resistivity tomography varies from the one obtained using ERTManager. Is this physically correct? 
I was under the impression that the ERT and IP inversions, even though IP is a derivative from resistivity, the resistivity tomographies should not be affected by the IP inversion, which is applied afterwards. However, I notice a difference in the resistivity images. 

I attach the resistivity image obtained with ERTManager for your reference:

![Captura](https://github.com/user-attachments/assets/d89b99cb-bb43-4229-b913-de614576e2d0)


Also, the ERT and IP images obtained with ERTIPManager:

![Captura](https://github.com/user-attachments/assets/bed31ce0-34a2-48d2-84cd-0bf6e4b6dcde)

Thanks again

**Comment by halbmy** [comment]
You're absolutely right, the result should be identical for the same keyword arguments. I will check this with an owl data set.

**Comment by halbmy** [comment]
`ERTIPManager` is directly inherited from `ERTManager` and takes the `.invert` or forwards the `invertDC` to the latter, so there should not be any difference. Can you please paste the code that leads to different results?

**Comment by Santi-Rebole-Canals** [comment]
**ERT Inversion code**
Data: 
[Pastillo_p2_sh_edited_topography_with_sensors.txt](https://github.com/user-attachments/files/17174505/Pastillo_p2_sh_edited_topography_with_sensors.txt)

After importing all the packages, and performing a linear interpolation, this is the code when using ERTManager:

`name_out = 'Pastillo_p2_sh_edited_topography_with_sensors.txt'`
`data2 = ert.load(name_out)`
`data2['k'] = ert.createGeometricFactors(data2, numerical=True) `
`data2['err'] = ert.estimateError(data2, relativeError=0.03, absoluteUError=5e-5)`
`mgr = ert.ERTManager(sr = False)`
`mod = mgr.invert(data2, lam=20, verbose=True, zWeight=0.5, paraDX = 5, quality=33.6, trapRatio=0.3)`
`mgr.showResult(mod, cMin=5, cMax=2000, logScale=True, coverage=1)`

**ERT and IP Inversion code**
Data with IP included: 
[Pastillo_p24sh_IP1_with_sensors.txt](https://github.com/user-attachments/files/17174514/Pastillo_p24sh_IP1_with_sensors.txt)

When using ERTIPManager, after importing all packages and performing a linear interpolation:

`nameip_out = 'Pastillo_p24sh_IP1_with_sensors.txt'`
`dataIP2 = ert.load(nameip_out)`
`dataIP2['k'] = ert.createGeometricFactors(dataIP2, numerical=True) `
`dataIP2['err'] = ert.estimateError(dataIP2, relativeError=0.03, absoluteUError=5e-5)`
`mgrIP2 = ert.ERTIPManager(dataIP2)`
`mgrIP2.invert(dataIP2, lam=20, verbose=True, zWeight=0.5, paraDX = 5, quality=33.6, trapRatio=0.3)`
`mgrIP2.showResult(cMin=5, cMax=2000, logScale=True, coverage=1)`
`mgrIP2.showIPModel()`

Greetings

**Comment by halbmy** [comment]
I don't quite understand. I thought it was about `ERTManager` and `ERTIPManager` providing different results for the same data but in your code you use two totally different data:
![image](https://github.com/user-attachments/assets/5e3d7985-7745-4b30-80fc-9719c42bf79d)
![image](https://github.com/user-attachments/assets/db260e2e-c7f3-4109-962c-011f67569f12)

the first one having a lot of systematic errors and therefore a bad data fit. If you use the `ERTManager` with the second data set, the inversion result is the same.

**Comment by ZhenluRen** [comment]
> **ERT Inversion code** Data: [Pastillo_p2_sh_edited_topography_with_sensors.txt](https://github.com/user-attachments/files/17174505/Pastillo_p2_sh_edited_topography_with_sensors.txt)
> 
> After importing all the packages, and performing a linear interpolation, this is the code when using ERTManager:
> 
> `name_out = 'Pastillo_p2_sh_edited_topography_with_sensors.txt'` `data2 = ert.load(name_out)` `data2['k'] = ert.createGeometricFactors(data2, numerical=True) ` `data2['err'] = ert.estimateError(data2, relativeError=0.03, absoluteUError=5e-5)` `mgr = ert.ERTManager(sr = False)` `mod = mgr.invert(data2, lam=20, verbose=True, zWeight=0.5, paraDX = 5, quality=33.6, trapRatio=0.3)` `mgr.showResult(mod, cMin=5, cMax=2000, logScale=True, coverage=1)`
> 
> **ERT and IP Inversion code** Data with IP included: [Pastillo_p24sh_IP1_with_sensors.txt](https://github.com/user-attachments/files/17174514/Pastillo_p24sh_IP1_with_sensors.txt)
> 
> When using ERTIPManager, after importing all packages and performing a linear interpolation:
> 
> `nameip_out = 'Pastillo_p24sh_IP1_with_sensors.txt'` `dataIP2 = ert.load(nameip_out)` `dataIP2['k'] = ert.createGeometricFactors(dataIP2, numerical=True) ` `dataIP2['err'] = ert.estimateError(dataIP2, relativeError=0.03, absoluteUError=5e-5)` `mgrIP2 = ert.ERTIPManager(dataIP2)` `mgrIP2.invert(dataIP2, lam=20, verbose=True, zWeight=0.5, paraDX = 5, quality=33.6, trapRatio=0.3)` `mgrIP2.showResult(cMin=5, cMax=2000, logScale=True, coverage=1)` `mgrIP2.showIPModel()`
> 
> Greetings

Dear Santi,

I browsed your question and found that the inversion graphs you made was very nice. I am curious about how you limit the inversion plot into the range of the inverted trapezoid. Would you be willing to share with me your code for the drawing part? Thank you.

Kind regards,
Zhenlu Ren

**Comment by halbmy** [comment]
The images just show the pseudosections by `data.show()` with no further settings.

**Comment by Santi-Rebole-Canals** [comment]
You are totally right.

The data I entered came from different survey types, albeit from the same profile. I thought they shared the same configurations. To test this, I used data from surveys that I was certain had the same configurations and it worked as you said: ERT graphs are the same independently from the type of Manager.

On the other side, I think @ZhenluRen is referring to the trapRatio function that is added when inverting. The code I used for representing is the same I provided in this issue.

Best regards.
